dev/io.openliberty.org.jboss.resteasy.common.ee10/src/org/jboss/resteasy/plugins/providers/multipart/Mime4JWorkaround.java

```diff
@@ -0,0 +1,287 @@
+            return new FileOutputStream(file);
+        }
+
+        private static final class TempFileStorageOutputStream extends StorageOutputStream
+        {
+            private File file;
+
+            private OutputStream out;
+
+            TempFileStorageOutputStream(final File file) throws IOException
+            {
+                this.file = file;
+                this.out = createFileOutputStream(file);
+            }
+
+            @Override
+            public void close() throws IOException
+            {
+                super.close();
+                out.close();
+            }
+
+            @Override
+            protected void write0(byte[] buffer, int offset, int length) throws IOException
+            {
+                out.write(buffer, offset, length);
+            }
+
+            @Override
+            protected Storage toStorage0() throws IOException
+            {
+                // out has already been closed because toStorage calls close
+                return new TempFileStorage(file);
+            }
+        }
+
+        private static final class TempFileStorage implements Storage
+        {
+
+            private File file;
+
+            private static final Set<File> filesToDelete = new HashSet<File>();
+
+            TempFileStorage(final File file)
+            {
+                this.file = file;
+            }
+
+            public void delete()
+            {
+                // deleting a file might not immediately succeed if there are still
+                // streams left open (especially under Windows). so we keep track of
+                // the files that have to be deleted and try to delete all these
+                // files each time this method gets invoked.
+
+                // a better but more complicated solution would be to start a
+                // separate thread that tries to delete the files periodically.
+
+                synchronized (filesToDelete)
+                {
+                    if (file != null)
+                    {
+                        filesToDelete.add(file);
+                        file = null;
+                    }
+
+                    for (Iterator<File> iterator = filesToDelete.iterator(); iterator.hasNext();)
+                    {
+                        File f = iterator.next();
+                        // Liberty start: Awaiting https://issues.redhat.com/projects/RESTEASY/issues/RESTEASY-3212
+                        boolean deleted;
+                        try {
+                            deleted = AccessController.doPrivileged(new PrivilegedExceptionAction<Boolean>() {
+                               @Override
+                               public Boolean run() throws Exception {
+                                   return f.delete();
+                                }
+                            });
+                         } catch (PrivilegedActionException pae)
+                         {      throw new RuntimeException(pae);    }
+
```

Comment:
```suggestion
                        boolean deleted = AccessController.doPrivileged(new PrivilegedAction<Boolean>() {
                               @Override
                               public Boolean run() {
                                   return f.delete();
                                }
                            });

```

In other places in this class they only do the doPriv if Java security is enabled.  May want to do the code similarly as what they did in createFileOutputStream or make a delete(File) private static method like createFileOutputStream and others are in the file.

Reply:
agreed